### PR TITLE
feat: modernize codebase

### PR DESCRIPTION
**Pull request by Maniktherana**
This pr:
 - migrates to TypeScript
 - adds shadcn ui
 - partially addresses #500 

Note: making changes to tailwind.config.ts or adding shadcn components requires a rebuild

### CODE REVIEW

**Comment by Maniktherana** [comment]
Changes so far:
 - new mobile navbar
 Screenshot 2024-02-19 at 1 35 15 PM

 - Status page
Screenshot 2024-02-19 at 1 35 44 PM

 - PR page
Screenshot 2024-02-19 at 1 36 11 PM

 - some tables now scroll horizontally for small screens instead of bunching up together

Screenshot 2024-02-19 at 1 37 17 PM

**Comment by Maniktherana** [comment]
This PR also breaks json rendering on the `/macrobench/queries/compare` route. Root cause is most likely not having access to the `json` class from `ReactJson`

before:
Screenshot 2024-02-19 at 2 12 06 PM

after:
Screenshot 2024-02-19 at 2 12 21 PM

File: website/src/assets/styles/tailwind.css [hunk 1]
```diff
@@ -5,7 +5,7 @@ Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
 You may obtain a copy of the License at
 
-    http://www.apache.org/licenses/LICENSE-2.0
+http://www.apache.org/licenses/LICENSE-2.0
```
Comment (frouioui) [inline website/src/assets/styles/tailwind.css:8]: nit, change not needed

File: website/src/assets/styles/tailwind.css [hunk 2]
```diff
@@ -18,6 +18,88 @@ limitations under the License.
...
+}
+
+/* @layer base {
+  * {
+    @apply border-border;
+  }
+  body {
+    @apply bg-background text-foreground;
+  }
+} */
```
Comment (frouioui) [inline website/src/assets/styles/tailwind.css:101]: If not needed in the future, we should remove this as the base layer already exist right below

File: website/src/common/Navbar.tsx [hunk 3]
```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (frouioui) [inline website/src/common/Navbar.tsx:2]:
```suggestion
Copyright 2024 The Vitess Authors.
```

File: website/src/components/mode-toggle.tsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (frouioui) [inline website/src/components/mode-toggle.tsx:2]:
```suggestion
Copyright 2024 The Vitess Authors.
```

File: website/src/components/theme-provider.tsx [hunk 5]
```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2023 The Vitess Authors.
+
```
Comment (frouioui) [inline website/src/components/theme-provider.tsx:3]:
```suggestion
Copyright 2024 The Vitess Authors.

```

File: website/src/components/ui/button.tsx [hunk 6]
```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (frouioui) [inline website/src/components/ui/button.tsx:2]:
```suggestion
Copyright 2024 The Vitess Authors.
```

File: website/src/components/ui/dropdown-menu.tsx [hunk 7]
```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (frouioui) [inline website/src/components/ui/dropdown-menu.tsx:2]:
```suggestion
Copyright 2024 The Vitess Authors.
```

File: website/src/components/ui/sheet.tsx [hunk 8]
```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (frouioui) [inline website/src/components/ui/sheet.tsx:2]:
```suggestion
Copyright 2024 The Vitess Authors.
```

File: website/src/contexts/GlobalContext.tsx [hunk 9]
```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (frouioui) [inline website/src/contexts/GlobalContext.tsx:2]:
```suggestion
Copyright 2024 The Vitess Authors.
```

File: website/src/pages/ForeignKeysPage/components/Hero.tsx [hunk 10]
```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (frouioui) [inline website/src/pages/ForeignKeysPage/components/Hero.tsx:2]:
```suggestion
Copyright 2024 The Vitess Authors.
```

File: website/src/pages/StatusPage/components/Hero.tsx [hunk 11]
```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2023 The Vitess Authors.
```
Comment (frouioui) [inline website/src/pages/StatusPage/components/Hero.tsx:2]:
```suggestion
Copyright 2024 The Vitess Authors.
```

**Review by frouioui — COMMENTED** [review]
Globally it looks great, it is a great step toward better UI/UX. The navbar renders really well on mobile, certain elements on the home page look cleaner too. Thank you!

I tested the query comparator page, and it works fine for me with production data.

> Note: making changes to tailwind.config.ts or adding shadcn components requires a rebuild

This sounds fine to me, the deploy script rebuilds all components.

---- 

As of right now I cannot merge the Pull Request due to the following points:
- On Safari the website does not render well at all. It, almost, only shows white pages when the dark mode is turned on. I would advise you to test the PR on different browsers and see how they behave.

- The page "Daily" causes a few issues. The background of the graphs are not showing up, and the summary cards (when selected) are highlighted with a dark/white color. While we don't have to necessarily focus on improving the UI yet, I think it is important to keep parity with the old UI while moving to Shadcn and completing the migration to TS.

Before:
image
image


After:
image
image

**Comment by Maniktherana** [comment]
> On Safari the website does not render well at all. It, almost, only shows white pages when the dark mode is turned on. I would advise you to test the PR on different browsers and see how they behave.

This has been fixed, the issue was a typo in `tailwind.css`
```diff
.dark {
-    --background: 0 0% 0;
+    --background: 0 0% 0%;
```

**Comment by Maniktherana** [comment]
>The page "Daily" causes a few issues. The background of the graphs are not showing up, and the summary cards (when selected) are highlighted with a dark/white color.

graphs are visible now on hovering summary cards

**Comment by Maniktherana** [comment]
> After: image

I've just pushed a potential fix for this but I'm going in blind. I can't see graphs yet on my dev build but the fix involves just using the correct css color variable.

**Comment by frouioui** [comment]
> I've just pushed a potential fix for this but I'm going in blind. I can't see graphs yet on my dev build but the fix involves just using the correct css color variable.

It looks great now:

image

**Pull request merged**